addons/stock/models/stock_move.py

```diff
@@ -1030,7 +1030,10 @@ def _action_confirm(self, merge=True, merge_into=False):
         self._push_apply()
         self._check_company()
         if merge:
-            return self._merge_moves(merge_into=merge_into)
+            res = self._merge_moves(merge_into=merge_into)
```

Comment:
```python3
moves = self
if merge:
    moves = self._merge_moves(merge_into=merge_into)
moves.filtered(lambda m: m._should_bypass_reservation() and m.state == 'confirmed')._action_assign()
return moves
```
Just in case _action_confirm(merge=False) is called on a move that bypass reservation.